### ISSUE TITLE
Debugging broken after MacOS Mojave Install

### ISSUE THREAD

**Issue by tfnaylor** [derekparker/delve#1398]
After installing MacOS Mojave and all updates, I can no longer debug from a go test file. I get the following error in the debug terminal (Visual Studio Code Version 1.28.2):

could not launch process: debugserver or lldb-server not found: install XCode's command line tools or lldb-server

I have re-installed the updated xcode and also ran xcode-select -- install

I have updated delve using  go get -u github.com/derekparker/delve/cmd/dlv

I am using go 1.11 (worked before Mojave install) on MacOS Mojave 10.14.1 (18B75).
I have a Macbook Pro Retina 2016

**Comment by aarzilli** [comment]
We look for debugserver at `/Library/Developer/CommandLineTools/Library/PrivateFrameworks/LLDB.framework/Versions/A/Resources/debugserver` if mojave moved it somewhere else we'd need to know where it is.

**Comment by tfnaylor** [comment]
The /Library/Developer/CommandLineTools directory is essentially empty. See attached screenshot.
<img width="1171" alt="screenshot 2018-11-03 at 07 52 37 pm" src="https://user-images.githubusercontent.com/1755292/47956276-31eb4b80-dfa2-11e8-8f51-2ee2b59b2830.png">

**Comment by tfnaylor** [comment]
After more banging at this, it seems that the Mojave installation removed the command line tools altogether and you can only install full xcode from the xcode-select --install command. Attempting to install the command line tools alone gives you a popup dialog like this:

<img width="501" alt="screenshot 2018-11-03 at 08 09 20 pm" src="https://user-images.githubusercontent.com/1755292/47956415-6a8c2480-dfa4-11e8-9a4a-1c191937bb86.png">

**Comment by tfnaylor** [comment]
Problem solved. It seems you have to download the updated command line tools from the Apple developer website (xcode-select --install) is broken for the latest MacOS Mojave version (10.14.1).

https://developer.apple.com/download/more/

After downloading and installing the tools from the web site, debugging is working again.

**Comment by aarzilli** [comment]
Glad you figured it out.

**Comment by acardero** [comment]
At this point in time `xcode-select --install` is working fine and correctly fixes the error on the debugger.

**Comment by jongear** [comment]
can also confirm that `xcode-select --install` solved my debugging issues with Go 1.12.5 on Mojave

**Comment by gflarity** [comment]
I had to delete/mv my old developer tools otherwise it complained it was already installed.
```
rm -f /Library/Developer/CommandLineTools 
```
then install works again:
```
xcode-select --install
```

However I still get the error:

```
error: failed to launch process /Library/Developer/CommandLineTools/Library/PrivateFrameworks/LLDB.framework/Versions/A/Resources/debugserver: No such file or directory
```

**Comment by lzantal** [comment]
This issue helped me as well after all these months. Mojave upgrade wiped my Xcode cmd tools 😳
Running xcode-select --install worked like a charm. I did have to fully restart vscode for debugging to start working.

**Comment by Arnie97** [comment]
Thanks, this solution also applies to macOS Catalina install / upgrade. I just upgraded my mac from Mojave to Catalina and got exactly the same issue.

**Comment by aallbrig** [comment]
Just as another datapoint: running `xcode-select --install` after doing a macOS Catalina (`10.15.4`) upgrade resolve an issue where Goland (`2020.1.2`) was unable to attach to the debugger service, thereby rendering the debug functionality unusable.

Goland is once again able to attach to the debugger service and I'm happily returning to debugging go code 🥳

**Comment by tclemos** [comment]
Here we are again with MacOS Big Sur 11.2 (20D64).
Just updated and it broke my VSCode debugging.
`xcode-select --install` fixed the problem
`go version go1.15.7 darwin/amd64`

**Comment by mghifariyusuf** [comment]
`xcode-select --install` still work fine for me!
Updated to Big Sur and have go version go1.16.6 darwin/amd64

**Comment by ganesh-getweave** [comment]
`xcode-select --install` fixed the problem after updating to Monterey 12.5.1.

**Comment by brunoluizkatz** [comment]
`xcode-select --install` fixed the problem after updating to Venture 13.0.1 (22A400)

**Comment by axilis-marko** [comment]
I have upgraded to Venture 13.1 (22C65) and I'm getting `Can’t install the software because it is not currently available from the Software Update server.` after running `xcode-select --install`. The https://developer.apple.com/download/more/ is asking for Apple ID sign in.
